### PR TITLE
Allow a person to (somewhat) automatically deploy & update IRC Hooky without needing to run a bunch of commands manually

### PR DESCRIPTION
**Pull request by marvinpinto**
This deployment script is is not as hardened as I would like, but it will have to do for now.

This is a lot cleaner than having to manually copy/paste instructions, while having to pay attention to which environment variables were set, etc.

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -16,6 +13,30 @@ There very well might be! But unfortunately this isn't it. The complexity of
...
+
+You are than welcome to use `AWS console`__, `aws cli`__, or any tool that you
+are more comfortable with.
+
+__ https://console.aws.amazon.com/console
+__ https://aws.amazon.com/cli
+
+The reasoning behind ``deploy.py`` versus writing out the instructions manually
+is because the latter got unweildy fast. Iterating on an environment where one
+has to copy/paste commands into a terminal window got combersome and I decided
```
Comment (margueritepd) [inline docs/faq.rst]: `cumbersome`

File: scripts/deploy.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+        client = boto3.client('iam')
+        self.lambda_policy_arn = ""
+        try:
+            response = client.create_policy(
+                PolicyName=policy_name,
+                PolicyDocument=json.dumps(policy_doc)
+            )
+            self.lambda_policy_arn = response['Policy']['Arn']
+            self.log.info("Policy %s successfully created." % self.lambda_policy_arn)  # NOQA
+        except botocore.exceptions.ClientError:
```
Comment (margueritepd) [inline scripts/deploy.py:71]: Is this the only ClientError possible during this scenario? you might hate yourself later if there's another possible clienterror, but your logging claims that it's caused by the policy already existing. Is there an error code you could validate?

**Comment by margueritepd** [comment]
:computer: :heavy_plus_sign: :heavy_plus_sign:
:book: :heavy_minus_sign: :heavy_minus_sign: 
:dog:

File: scripts/deploy.py (hunk 2 above)
Comment (marvinpinto) [inline scripts/deploy.py:71]: I have found boto's exception handling to be not so great :disappointed: 

You bring up a good point and I'm going to keep this in mind.

(thank you for reading this PR :smile:)

**Pull request merged**